let's modernize this thing

twolife opened the pull request:
I ported the code from GConf to GSettings.
(It's not a piece of art but it works)

tatankat commented on configure.ac:
```diff
@@ -26,70 +26,36 @@ AC_CHECK_HEADERS([ctype.h dirent.h errno.h locale.h signal.h \
 
-# Initialise i18n
-AM_GNU_GETTEXT_VERSION([0.17])
-AM_GNU_GETTEXT([external])
```
Without these, running autogen fails (on a fresh checkout) with following output:
* at the beginning
```
autopoint: *** Missing version: please specify in configure.ac through a line 'AM_GNU_GETTEXT_VERSION(x.yy.zz)' the gettext version the package is using
autopoint: *** Stop.
```
* at the end
```config.status: error: cannot find input file: `po/Makefile.in.in'```

Did you remove these for a reason?

twolife replied:
I followed the advice from https://mail.gnome.org/archives/commits-list/2012-March/msg04458.html
GETTEXT and all i18n stuffs are blackmagic to me, but if i don't remove those 2 lines, i'm stuck with `config.status: error: po/Makefile.in.in was not created by intltoolize.`

tatankat commented:
Nice, I am happy this has been picked up.

Good work on removing gconf. Some issues I encountered:
* it crashes when changing themes
* the setting changes are no longer reflected immediately (only after you restart the program, but we can merge without this)
* I had to compile the schema by hand, although this is defined in the Makefile (no clue why yet)
* I had to run it with the GSETTINGS_SCHEMA_DIR set as I do not install this in the root directories (no big problem, I don't know if this can be done automatically)

Another issue: the layout of the Partyline and Winlist has been horizontally squashed together.
About the sound: I never played with sound. I compiled it now with sound enabled, but did not hear anything. I did not look into this.

Thanks for the effort! I can merge without any changes and then do some fixes myself or you can do it in the pull request. Your choice.

twolife commented:
> * it crashes when changing themes

oops sorry about that. i fixed it.
 
> * the setting changes are no longer reflected immediately (only after you restart the program, but we can merge without this)

Right, i will try to fix that "soon" :-)

> * I had to compile the schema by hand, although this is defined in the Makefile (no clue why yet)
> * I had to run it with the GSETTINGS_SCHEMA_DIR set as I do not install this in the root directories (no big problem, I don't know if this can be done automatically)
>
> Another issue: the layout of the Partyline and Winlist has been horizontally squashed together.
> About the sound: I never played with sound. I compiled it now with sound enabled, but did not hear anything. I did not look into this.

I never used the sound either, i just fixed the autoconf stuff to make it build, i did not really test.

> Thanks for the effort! I can merge without any changes and then do some fixes myself or you can do it in the pull request. Your choice.

Please merge, i will make new pull request when/if i fixed something more :-)

tatankat commented:
> > * it crashes when changing themes
>
> oops sorry about that. i fixed it.
>
thanks
> > * the setting changes are no longer reflected immediately (only after you restart the program, but we can merge without this)
>
> Right, i will try to fix that "soon" :-)
>
soon is good :)
> > * I had to compile the schema by hand, although this is defined in the Makefile (no clue why yet)

fixed in master

> > * I had to run it with the GSETTINGS_SCHEMA_DIR set as I do not install this in the root directories (no big problem, I don't know if this can be done automatically)

will document this or look into this soon

> > Another issue: the layout of the Partyline and Winlist has been horizontally squashed together.

will fix this very soon

> > About the sound: I never played with sound. I compiled it now with sound enabled, but did not hear anything. I did not look into this.
>
> I never used the sound either, i just fixed the autoconf stuff to make it build, i did not really test.
> 
probably something to do less "soon"

> > Thanks for the effort! I can merge without any changes and then do some fixes myself or you can do it in the pull request. Your choice.
>
> Please merge, i will make new pull request when/if i fixed something more :-)
>
done

I also tried to move completely to gettext as proposed in https://wiki.gnome.org/Initiatives/GnomeGoals/GettextMigration I hope you no longer receive the error.
